File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/ESurfaceBrightDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESurfaceBrightDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd9
            AppTheme.STANDARD -> R.color.cstd9
            AppTheme.BLUE -> R.color.cbd9
            AppTheme.GREEN -> R.color.cgd9
            AppTheme.RED -> R.color.crd9
            AppTheme.NOTHING -> R.color.cnod9
        }
    }
}

File: lib/layer2_core/dark/e_surface_bright_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class ESurfaceBrightDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd9;
      case 'standard': return cstd9;
      case 'blue': return cbd9;
      case 'green': return cgd9;
      case 'red': return crd9;
      case 'nothing': return cnod9;
      default: return cstd9;
    }
  }
}
